hydra-node/test/Hydra/Chain/Direct/TxSpec.hs

```diff
@@ -61,12 +62,19 @@ spec =
         headSeedToTxIn headSeed === Just txIn
           & counterexample (show headSeed)
 
-    describe "HeadId (cardano)" $
+    describe "HeadId (cardano)" $ do
       prop "headIdToPolicyId . mkHeadId === id" $ \pid -> do
         let headId = mkHeadId pid
         headIdToPolicyId headId === Just pid
           & counterexample (show headId)
 
+      prop "curencySymbolToHeadId . headIdToCurrencySymbol === id" $ \txIn -> monadicIO $ do
+        let headId = mkHeadId $ headPolicyId txIn
```

Comment:
You do not need `txIn`, why not generate an arbitrary `HeadId` directly?

Could be that `arbitrary` for `HeadId` is not (well-)defined? If that is the case, fixing this would be great.